bindings/azure/servicebusqueues/servicebusqueues.go

```diff
@@ -156,31 +153,23 @@ func (a *AzureServiceBusQueues) Invoke(req *bindings.InvokeRequest) (*bindings.I
 	}
 	defer client.Close(context.Background())
 
-	msg := servicebus.NewMessage(req.Data)
-	if val, ok := req.Metadata[id]; ok && val != "" {
-		msg.ID = val
-	}
-	if val, ok := req.Metadata[correlationID]; ok && val != "" {
-		msg.CorrelationID = val
-	}
-
-	ttl, ok, err := contrib_metadata.TryGetTTL(req.Metadata)
+	msg, err := asbmessage.NewASBMessageFromMessageWithMetadata(req)
 	if err != nil {
 		return nil, err
 	}
 
-	if ok {
-		msg.TTL = &ttl
-	}
-
 	return nil, client.Send(ctx, msg)
 }
 
 func (a *AzureServiceBusQueues) Read(handler func(*bindings.ReadResponse) ([]byte, error)) error {
 	var sbHandler servicebus.HandlerFunc = func(ctx context.Context, msg *servicebus.Message) error {
-		_, err := handler(&bindings.ReadResponse{
-			Data:     msg.Data,
-			Metadata: map[string]string{id: msg.ID, correlationID: msg.CorrelationID, label: msg.Label},
+		mmd, err := asbmessage.NewMessageWithMetadataFromASBMessage(msg)
+		if err != nil {
```

Comment:
The change in semantics here is tricky: functionally, nothing has changed in this implementation since `NewMessageWithMetadataFromASBMessage` never actually returns errors today, but you're suggesting that the Dapr _should_ enforce property mapping errors here. Some future dev may assume has always been the case and is well-handled when this could be a behavior breaking change when exercised for the first time.

My understanding of bindings is that they are generally dumb pipes as much as possible, with the responsibility on the app to handle any errors or issues, whereas pubsub is allowed to be more opinionated. As such, I would suggest that `NewMessageWithMetadataFromASBMessage` does not return errors, and any future metadata validation is the responsibility of the consuming component (i.e. pubsub component or bindings consumer).

Reply:
Right now as you said, there is no error returned, I will remove that field completely and will also do it for the other function.